app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManager.kt

```diff
@@ -0,0 +1,23 @@
+package org.p2p.wallet.swap.jupiter.statemanager
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collectLatest
+
+class SwapStateManager(private val handlers: Set<SwapStateHandler>) {
```

Comment:
the logic is same like in org.p2p.wallet.auth.repository.RestoreUserResultHandler